### PR TITLE
Creating a data model

### PR DESCRIPTION
**Pull request by MAleemH**
**I have done the following in this PR:**

- [x] Created a new rails app
- [x] Added required linters
- [x] Build schema according to provided ERD by using rails migrations

### CODE REVIEW

File: db/migrate/20230503132014_create_posts.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class CreatePosts < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.references :user, null: false, foreign_key: true
```
Comment (rotimiazeez) [inline db/migrate/20230503132014_create_posts.rb]: - - Great work on your migration files, however, I would recommend you use `author_id` when referencing the `user` in your posts table as it is in the ERD diagram  👍🏽

File: db/migrate/20230503133754_create_comments.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class CreateComments < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.references :user, null: false, foreign_key: true
+      t.references :post, null: false, foreign_key: true
+      t.text :text
+
```
Comment (rotimiazeez) [inline db/migrate/20230503133754_create_comments.rb]: - Great work on your migration files, however, I would recommend you use `author_id` when referencing the `user` and `post_id` when referencing `post` in your comments table as it is in the ERD diagram 😉

File: db/migrate/20230503133357_create_likes.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class CreateLikes < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+      t.references :user, null: false, foreign_key: true
+      t.references :post, null: false, foreign_key: true
+
```
Comment (rotimiazeez) [inline db/migrate/20230503133357_create_likes.rb]: - Great work on your migration files, however, I would recommend you use `author_id` when referencing the `user` and `post_id` when referencing `post` in your likes table as it is in the ERD diagram 😉

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi @MAleemH 👍🏽,
## Good job so far!
### To Highlight 👇🏽
- Good job adding the migration file 👏🏽 
- No Linter error ✔️ 
- Good PR description 👍🏽 
- Good commit message(s) 😉
- Descriptive README 📖
You still need to work on some issues to prepare your project for the final evaluation, but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR WORDS>


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you consider them as they can improve your code._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag (@rotimiazeez) me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

File: README.md [hunk 4]
```diff
@@ -1 +1,260 @@
...
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+> Link will be provided soon.
+
+- [Live Demo Link](#)
```
Comment (danmainah) [inline README.md:108]: - [ ] [optional] @MAleemH great work on your documentation, however, I would recommend you remove the live demo section and add it when you finally deploy the app

**Review by danmainah — APPROVED** [review]
Hi @MAleemH , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ No linter errors
✔️ Giflow was correctly used
✔️  Good design on your schema

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**